src/online.py

```diff
@@ -0,0 +1,332 @@
+
+    def crack(self, hash):
+        try:
+            out = requests.get(
+                f"https://md5.gromweb.com/?md5={hash[0]}", timeout=hash[3]
+            ).text
+            text = "".join(
+                out.split(
+                    '<input class="field" id="form_string_to_hash_string" type="search" name="string" value="'
+                )[1]
+            ).split('"')[0]
+            if not text:
+                return "Failed"
+            return text
+        except:
+            return "Not connected"
+
+
+class sha1_grom:
+
+    supports = set(["sha-1"])
+
+    def crack(self, hash):
+        try:
+            out = requests.get(
+                f"https://sha1.gromweb.com/?hash={hash[0]}", timeout=hash[3]
+            ).text
+            text = "".join(
+                out.split(
+                    '<input class="field" id="form_string_to_hash_string" type="search" name="string" value="'
+                )[1]
+            ).split('"')[0]
+            if not text:
+                return "Failed"
+            return text
+        except:
+            return "Not connected"
+
+
+class opcrack:
+
+    supports = set(["ntlm", "lm"])
+
+    def crack(self, hash):
+        try:
+            burp0_url = "https://cracker.okx.ch:443/crack"
+            burp0_headers = {
+                "Connection": "open",
+                "Accept": "application/json, text/javascript, */*; q=0.01",
+                "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36",
+                "Content-Type": "application/json",
+                "Origin": "https://www.objectif-securite.ch",
+                "Sec-Fetch-Site": "cross-site",
+                "Sec-Fetch-Mode": "cors",
+                "Sec-Fetch-Dest": "empty",
+                "Referer": "https://www.objectif-securite.ch/",
+                "Accept-Encoding": "gzip, deflate",
+                "Accept-Language": "en-GB,en-US;q=0.9,en;q=0.8",
+            }
+            burp0_json = {"value": hash[0]}
+
+            text = requests.post(
+                burp0_url, headers=burp0_headers, json=burp0_json, timeout=hash[3]
+            ).text
+
+            return "".join(text.split('":"')[1]).split('"}')[0]
+        except:
+            return "Not connected"
+
+
+class rainbow_tabels:
+
+    supports = set(["sha-256", "md5", "md4", "NTLM"])
+
+    def crack(self, hash):
+        try:
+            out = requests.get(
+                f"https://hashdecryption.com/decrypt.php?str={hash[0]}&send=Submit",
+                timeout=hash[4],
+            ).text
+            return "".join(out.split("</b> is <b>")[1]).split("</b><br>")[0]
+        except:
+            return "Failed"
+
+
+# Bug Fixing - API / Domain is currently down
+
+
+class hashsorg:
+
+    supports = set(["md5", "ntlm", "sha-1", "md4"])
+    moduels = ["requests"]
+    offline = False
+
+    # How do we implment the API key?

```

Comment:
github issue, del, and use the config file which should be implemented in main :)